### PR TITLE
Add description for populateMatrix()

### PR DESCRIPTION
**Pull request by pozdnyakov**
***
[Preview](https://s3.amazonaws.com/pr-preview/pozdnyakov/orientation-sensor/populateMatrix_description.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/orientation-sensor/670182e...pozdnyakov:1de6e30.html)

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -182,21 +193,55 @@ To <dfn>Construct a OrientationSensor Object</dfn> the user agent must invoke th
...
         1.  Let |quaternion| be the value of [=latest reading=]["quaternion"]
         1.  If |quaternion| is `null`, abort these steps.
         1.  Copy |quaternion| into |targetBuffer|
 </div>
 
 ### OrientationSensor.populateMatrix() ### {#orientationsensor-populatematrix}
 
-<div algorithm="to populate matrix">
-The {{OrientationSensor/populateMatrix()}} method must run these steps or their [=equivalent=]:
+The {{OrientationSensor/populateMatrix()}} method  populates the given buffer with WebGL-compatible rotation matrix
```
Comment (kenchris) [inline index.bs]: with *a*... you should state whether it is column major or row major.

File: index.bs [hunk 2]
```diff
@@ -182,21 +193,55 @@ To <dfn>Construct a OrientationSensor Object</dfn> the user agent must invoke th
...
+ - X = V<sub>x</sub> * sin(θ/2).
+ - Y = V<sub>y</sub> * sin(θ/2).
+ - Z = V<sub>z</sub> * sin(θ/2).
+
+<div algorithm="to populate WebGL-compatible matrix">
+The {{OrientationSensor/populateMatrix()}} method must run these steps or their [=equivalent=]:
+    1.  If |targetBuffer| is not of type {{Float32Array}}, [=throw=] a
+        "{{SyntaxError!!exception}}" {{DOMException}} and abort these steps.
+    1.  If |targetBuffer| is of type {{Float32Array}} with a size less than sixteen, [=throw=] a
+        "{{SyntaxError!!exception}}" {{DOMException}} and abort these steps.
```
Comment (kenchris) [inline index.bs]: Is SyntaxError really the best error kind

Comment (pozdnyakov) [inline index.bs]: what would be your proposal? RangeError?

Comment (alexshalamov) [inline index.bs]: I would propose to use TypeError. https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror

> TypeError is used to indicate an unsuccessful operation when none of the other NativeError objects are an appropriate indication of the failure cause.

wdyt?

File: index.bs (hunk 1 above)
Comment (pozdnyakov) [inline index.bs]: done

**Comment by pozdnyakov** [comment]
@kenchris , @alexshalamov PTAL

File: index.bs [hunk 3]
```diff
@@ -182,21 +193,55 @@ To <dfn>Construct a OrientationSensor Object</dfn> the user agent must invoke th
...
+    1.  If the value of |sensor_instance|.[=[[state]]=] is not <a enum-value>"activated"</a>,
+        [=throw=] an "{{InvalidStateError!!exception}}" {{DOMException}} and abort these steps.
+    1.  Run these sub-steps:
+        1.  Let |quaternion| be the value of [=latest reading=]["quaternion"]
+        1.  If |quaternion| is `null`, abort these steps.
```
Comment (pozdnyakov) [inline index.bs]: doesn't it come from model description?

File: index.bs [hunk 4]
```diff
@@ -182,21 +193,55 @@ To <dfn>Construct a OrientationSensor Object</dfn> the user agent must invoke th
...
+        1.  Let |W| be the value of |quaternion|[0].
```
Comment (pozdnyakov) [inline index.bs]: yeah, I'll change that

**Pull request merged**